src/components/gcds-site-menu/gcds-site-menu.tsx

```diff
@@ -0,0 +1,310 @@
+    //           displayMenuChildren[i].setAttribute('data-toolbar-child', '');
+    //         }
+    //       }
+    //     }
+
+    //     mainMenus[1].setAttribute('data-toolbar-parent', '');
+    //     for (var i =0; i < mainMenus[1].children.length; i++) {
+    //       mainMenus[1].children[i].setAttribute('data-toolbar-child', '');
+    //     }
+    //   }
+    // }
+  }
+
+  async componentWillLoad() {
+    // Define lang attribute
+    if(!this.el.getAttribute("lang")) {
+      if (document.documentElement.getAttribute("lang") == "en" || !document.documentElement.getAttribute("lang")) {
+        this.lang = "en";
+      } else {
+        this.lang = "fr";
+      }
+    } else if(this.el.getAttribute("lang") == "en") {
+      this.lang = "en";
+    } else {
+      this.lang = "fr";
+    }
+
+    this.validateDesktopLayout(this.menuDesktopLayout);
+    this.validateMobileLayout(this.menuMobileLayout);
+
+
+    // Add required attributes to slotted <ul>
+    await this.configureMenu();
+
+  }
+
+  componentDidLoad() {
+    var menus = [];
+    // Add ULs to shadow DOM
+    var elementChildren = this.el.children;
+    // Loop through slotted elements
+    for (var i = 0; i < elementChildren.length; i++) {
+      // Act only on ul
+      if (elementChildren[i].nodeName == "UL") {
+        menus = menus.concat(elementChildren[i]);
+      }
+    }
+
+    for (var i = 0; i < menus.length; i++) {
+      var container = this.el.shadowRoot.querySelector("[data-h2-menu-container]");
+      container.insertBefore(menus[i], container.querySelector('[data-optional-right]'));
+    }
+
+    if (this.menuDesktopLayout == "sidebar") {
+      // this.el.shadowRoot.querySelector("nav").style.setProperty(`--page-top-offset`, `${this.el.offsetTop}px`);
+    }
+
+    var hostElement = this.el;
+    var mobileLayout = this.menuMobileLayout;
+    const mediaQuery = window.matchMedia('screen and (min-width: 64em)');
+
+    // Check if loaded in mobile size
+    if (!mediaQuery.matches && mobileLayout == "drawer") {
+      document.querySelector("body").style.paddingBottom = "3rem";
+    }
+
+    // Register event listener
+    mediaQuery.addEventListener("change", function(e){
+      if (e.matches) {
+
+        // Set mobile menu to default if window size changes to desktop
+        var elementNav = hostElement.shadowRoot.querySelector('nav');
+        var mobileTrigger = hostElement.shadowRoot.querySelector('[data-h2-mobile-menu-trigger]');
+        if (elementNav.classList.contains("h2-mobile-menu-active")) {
+          elementNav.classList.remove("h2-mobile-menu-active");
+          mobileTrigger.classList.remove("h2-active");
+          mobileTrigger.setAttribute("aria-expanded", "false");
+          document.body.style.removeProperty("overflow");
+        }
+
+        if (mobileLayout == "drawer") {
+          document.querySelector("body").style.removeProperty("padding-bottom");
+        }
+
+      } else {
+        // var mainMenus = [];
```

Comment:
Same comment as above: Is this code needed for future functionality or can we remove it?

Reply:
Same as above. Will hopefully be implemented with toolbar layout